crypto/asn1/ameth_lib.c

```diff
@@ -216,6 +216,17 @@ EVP_PKEY_ASN1_METHOD *EVP_PKEY_asn1_new(int id, int flags,
             goto err;
     }
 
+    /*
+     * One of the following must be true:
+     *
+     * pem_str == NULL AND ASN1_PKEY_ALIAS is set
+     * pem_str != NULL AND ASN1_PKEY_ALIAS is clear
```

Comment:
So the "good case" is  `(pem_str == NULL) == ((flags & ASN1_PKEY_ALIAS) != 0))` ...